### ISSUE TITLE
Weird artifacts when testing on Kodak dataset

### ISSUE THREAD

**Issue by amnesiack** [pkuxmq/Invertible-Image-Rescaling#13]
Thanks for sharing the great work!

I had a test of IRN_x2 on the Kodak dataset (http://r0k.us/graphics/kodak/), which often adoped in multiple image-related tasks.
I saw some weird green shadow-like artifacts in many of the reconstructed images. both in LR and the reconstructed ones (like images shown below). Since the Kodak is also natural image dataset, I don't expect it has special characteristics or distributions compared to set5, set14, or DIV2K, which will lead to these artifacts. Did I ignore something or make something wrong? Do you observe similar artifacts in your experiments?

![kodim05_LR](https://user-images.githubusercontent.com/25249388/92610422-f6b1b600-f2b7-11ea-9351-e7cca490e210.png)
![kodim05](https://user-images.githubusercontent.com/25249388/92610294-d7b32400-f2b7-11ea-97dc-727916e0a7c8.png)
![kodim23_LR](https://user-images.githubusercontent.com/25249388/92610390-ec8fb780-f2b7-11ea-9326-6ae6d62f98a4.png)
![kodim23](https://user-images.githubusercontent.com/25249388/92610463-00d3b480-f2b8-11ea-970e-2831f8a3a7d5.png)

**Comment by pkuxmq** [comment]
Hi, thanks for pointing out it. I have verified it and it is interesting that IRN_x4 does not have such artifacts on Kodak, but IRN_x2 has artifacts on 12 images (half of Kodak). Besides, I tested another IRN_x2 model (I finetuned the pretrained model with L1 loss for LR guidance and add random noise on LR images before upscale during training), the artifacts are largely reduced but about 5 images still have few artifacts. I did not observe such artifacts on Set5, Set14, B100, Urban100, and DIV2K for all scales. It may be related with network architecture, training methods, and training dataset. We will further explore it.

**Comment by charliewang789** [comment]
I also see similar artifact on many images even with DiV2k. sounds like author did not see it?

**Comment by pkuxmq** [comment]
Hi @charliewang789 , what kind of artifact do you observe and what's the test setting? We do not observe artifacts on val_DIV2K,  Could you please provide detailed information, e.g. which image in DIV2K and which scale? Thanks!

**Comment by charliewang789** [comment]
not all epoc, only certain time, it appears. 2x scaling, example below.
![0894x2_forwLR_55000](https://user-images.githubusercontent.com/30919727/104271413-b9208d80-544f-11eb-8813-9123447a9028.png)

**Comment by steven8274** [comment]
Yeah, I met this problem also.When I use a screenshot image as HR input.The output LR and SR image all have added green color like this:
**SR:**
![screen_mac_3](https://user-images.githubusercontent.com/18027041/162353788-2b058152-cbfe-493e-ae3c-ea3892042796.png)
**LR**
![screen_mac_3_LR](https://user-images.githubusercontent.com/18027041/162354058-a600baba-3af2-4dab-9850-6fe663596d2d.png)
**Ground truth HR image:**
![screen_mac_3_GT](https://user-images.githubusercontent.com/18027041/162353850-5b63d748-7d8d-45ea-ad35-16d883765952.png)

I trained the model with IRN_x4 mode and ran 500000 iterators as default config.But the problem is gone if I use the model after 300000 iterators.

**Comment by pkuxmq** [comment]
The artifacts are likely to be caused by unrobustness of LR generation for some unseen distributions. One solution is also to finetune models on several samples of the target distribution. For example, if we finetune the model for several epochs on Kodak, the artifacts on Kodak with pretrained models can be removed. Another method is to increase the loss weight for LR guidance during training. This may lead to a slight performance drop for HR reconstruction (around 0.1-0.2 dB if we increase the weight by 10 times), but the restriction for valid LR image is stronger. Also, there may be other methods to regularize the model from overfitting that require exploration.
